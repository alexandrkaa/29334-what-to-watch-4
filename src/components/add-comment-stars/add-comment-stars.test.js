import React from 'react';
import renderer from 'react-test-renderer';
import AddCommentStars from './add-comment-stars.jsx';

describe(`<AddCommentStars /> should match snapshot`, () => {
  const onRadioChange = jest.fn();
  const rating = 1;
  it(`<AddCommentForm /> should match snapshot not disabled`, () => {
    const tree = renderer
      .create(
          <AddCommentStars
            starsNumber={3}
            rating={rating}
            isDisabled={false}
            onRadioChange={onRadioChange}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`<AddCommentStars /> should match snapshot disabled`, () => {
    const tree = renderer
      .create(
          <AddCommentStars
            starsNumber={3}
            rating={rating}
            isDisabled={true}
            onRadioChange={onRadioChange}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
