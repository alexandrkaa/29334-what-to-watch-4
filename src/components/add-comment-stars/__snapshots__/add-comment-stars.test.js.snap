// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCommentStars /> should match snapshot <AddCommentForm /> should match snapshot not disabled 1`] = `
Array [
  <input
    checked={true}
    className="rating__input"
    disabled={false}
    id="star-1"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={1}
  />,
  <label
    className="rating__label"
    htmlFor="star-1"
  >
    Rating 1
  </label>,
  <input
    checked={false}
    className="rating__input"
    disabled={false}
    id="star-2"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={2}
  />,
  <label
    className="rating__label"
    htmlFor="star-2"
  >
    Rating 2
  </label>,
  <input
    checked={false}
    className="rating__input"
    disabled={false}
    id="star-3"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={3}
  />,
  <label
    className="rating__label"
    htmlFor="star-3"
  >
    Rating 3
  </label>,
]
`;

exports[`<AddCommentStars /> should match snapshot <AddCommentStars /> should match snapshot disabled 1`] = `
Array [
  <input
    checked={true}
    className="rating__input"
    disabled={true}
    id="star-1"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={1}
  />,
  <label
    className="rating__label"
    htmlFor="star-1"
  >
    Rating 1
  </label>,
  <input
    checked={false}
    className="rating__input"
    disabled={true}
    id="star-2"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={2}
  />,
  <label
    className="rating__label"
    htmlFor="star-2"
  >
    Rating 2
  </label>,
  <input
    checked={false}
    className="rating__input"
    disabled={true}
    id="star-3"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={3}
  />,
  <label
    className="rating__label"
    htmlFor="star-3"
  >
    Rating 3
  </label>,
]
`;
